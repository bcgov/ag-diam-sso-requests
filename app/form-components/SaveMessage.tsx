import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faCheck, faExclamationTriangle } from '@fortawesome/free-solid-svg-icons';
import { SaveMessage as SaveMessageInterface } from 'interfaces/form';
import styled from 'styled-components';
import { Grid as SpinnerGrid } from 'react-loader-spinner';

const Icon = styled.span`
  width: 30px;
`;

const SaveContainer = styled.div`
  display: flex;
`;

const StyledP = styled.p`
  margin-left: 10px;
`;

interface Props {
  saving?: boolean;
  saveMessage?: SaveMessageInterface;
}

export default function SaveMessage({ saving, saveMessage }: Props) {
  const icon = saveMessage?.error ? faExclamationTriangle : faCheck;
  return (
    <>
      <SaveContainer>
        <Icon>
          {saving ? (
            // @ts-ignore
            <SpinnerGrid color="#000" height={18} width={50} visible label="request-saving" />
          ) : (
            <FontAwesomeIcon style={{ color: '#006fc4' }} icon={icon} title="request-saved" />
          )}
        </Icon>
        <StyledP>{saveMessage?.content}</StyledP>
      </SaveContainer>
    </>
  );
}
