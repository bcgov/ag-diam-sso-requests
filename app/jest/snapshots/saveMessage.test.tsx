import SaveMessage from 'form-components/SaveMessage';
import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

it('should match the snapshot', () => {
  let { asFragment } = render(<SaveMessage saving={true} content="saving" />);
  expect(asFragment()).toMatchSnapshot();

  ({ asFragment } = render(<SaveMessage saving={true} content="saving" variant="error" />));
  expect(asFragment()).toMatchSnapshot();
});
