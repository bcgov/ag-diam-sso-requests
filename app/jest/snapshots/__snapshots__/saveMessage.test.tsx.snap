// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-ftvSup gqkVqB"
  >
    <span
      class="sc-iBkjds flrtOA"
    >
      <svg
        class="sc-hKMtZM eAShCR"
        data-testid="rotating-lines-svg"
        role="status"
        speed="0.75"
        stroke="grey"
        viewBox="0 0 48 48"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(0, 24, 24)"
          width="5"
        />
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(30, 24, 24)"
          width="5"
        />
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(60, 24, 24)"
          width="5"
        />
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(90, 24, 24)"
          width="5"
        />
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(120, 24, 24)"
          width="5"
        />
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(150, 24, 24)"
          width="5"
        />
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(180, 24, 24)"
          width="5"
        />
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(210, 24, 24)"
          width="5"
        />
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(240, 24, 24)"
          width="5"
        />
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(270, 24, 24)"
          width="5"
        />
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(300, 24, 24)"
          width="5"
        />
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(330, 24, 24)"
          width="5"
        />
      </svg>
    </span>
    <p
      class="sc-papXJ gTcYUq"
    >
      saving
    </p>
  </div>
</DocumentFragment>
`;

exports[`should match the snapshot 2`] = `
<DocumentFragment>
  <div
    class="sc-ftvSup gqkVqB"
  >
    <span
      class="sc-iBkjds flrtOA"
    >
      <svg
        class="sc-hKMtZM eAShCR"
        data-testid="rotating-lines-svg"
        role="status"
        speed="0.75"
        stroke="grey"
        viewBox="0 0 48 48"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(0, 24, 24)"
          width="5"
        />
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(30, 24, 24)"
          width="5"
        />
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(60, 24, 24)"
          width="5"
        />
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(90, 24, 24)"
          width="5"
        />
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(120, 24, 24)"
          width="5"
        />
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(150, 24, 24)"
          width="5"
        />
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(180, 24, 24)"
          width="5"
        />
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(210, 24, 24)"
          width="5"
        />
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(240, 24, 24)"
          width="5"
        />
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(270, 24, 24)"
          width="5"
        />
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(300, 24, 24)"
          width="5"
        />
        <polyline
          class="sc-eCYdqJ lpbYzG"
          points="24,12 24,4"
          transform="rotate(330, 24, 24)"
          width="5"
        />
      </svg>
    </span>
    <p
      class="sc-papXJ gTcYUq"
    >
      saving
    </p>
  </div>
</DocumentFragment>
`;
